pages/nakshatra.vue

```diff
@@ -0,0 +1,379 @@
+        <option value="3">{{nakGirl3}}</option>
+        </select>
+      </div>
+      </div>
+      <button
+          class="button is-info m-2"
+          type="button"
+          v-on:click="calculate()"
+          :disabled="!allset"
+        >
+          Calculate
+        </button>
+    </form>
+    <div v-if="valuesVisibility" class="p-4 mt-10 mb-10 content max-w-xl">
+        <table class="table is-striped is-bordered is-hoverable card">
+            <tr class="bg-green-400 text-gray-900">
+                <th>Kutas</th>
+                <th>Gained</th>
+                <th>Max</th>
+            </tr>
+            <tr v-for="(value, propertyName,index) in resp.kutas" :key="index">
+               <td>{{propertyName}}</td>
+               <td>{{value.gained}}</td>
+               <td>{{value.max}}</td>
+            </tr>
+            <tr class="has-text-weight-bold">
+                <td>Total</td>
+                <td>{{resp.totalKutasGained.gained}}</td>
+                <td>{{resp.totalKutasGained.max}}</td>
+            </tr>
+            <tr>
+                <td colspan="3">
+                Important Note: The star matching is just the first step, don't assume that only star matching is sufficient and fail to check the
+                 overall horoscope for other factors(childbirth, happy family life, divorce possibilities, progeny, financial status, etc).
+                </td>
+            </tr>
+        </table>
+    </div>
+  </section>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop } from 'vue-property-decorator'
+import { getCurrentPageUrl } from '~/mixins/AppUtils'
+
+@Component
+export default class NakshatraVue extends Vue {
+  @Prop({ type: Object, required: true })
+  rasiBoy = '0'
+
+  @Prop({ type: Object, required: true })
+  rasiGirl = '0'
+
+  @Prop({ type: Object, required: true })
+  nkb1 = ''
+
+  @Prop({ type: Object, required: true })
+  nkb2 = ''
+
+  @Prop({ type: Object, required: true })
+  nkb3 = ''
+
+  @Prop({ type: Object, required: true })
+  nkg1 = ''
+
+  @Prop({ type: Object, required: true })
+  nkg2 = ''
+
+  @Prop({ type: Object, required: true })
+  nkg3 = ''
+
+  @Prop({ type: Object, required: true })
+  nakshatraGirl = '1'
+
+  @Prop({ type: Object, required: true })
+  nakshatraBoy = '1'
+
+  @Prop({ type: Object, required: true })
+  resp = {
+    value: true
+  }
+
+  @Prop({ type: Object, required: true })
+  nakshatraFinal = 'p'
+
+  @Prop({ type: Boolean, required: true })
+  valuesVisibility = false
+
+  @Prop({ type: Object, required: true })
+  allset = false
+
+  @Prop({ type: Object, required: true })
+  isLoading = false
+
+  head () {
+    return {
+      titleTemplate: 'Nakshatra| Innovative Astro Solutions',
+      meta: [
+        { name: 'og:url', content: getCurrentPageUrl('/nakshatra') },
+        { name: 'twitter:url', content: getCurrentPageUrl('/nakshatra') }
+      ]
+    }
+  }
+
+  calcBoy () {
+    if (this.rasiBoy === '1') {
```

Comment:
Remove duplicating code between `calcBoy` and `calcGirl` . Create a reusable method that accepts rasi as input and return an array of 3 nakshatra values. Then call the method and assign this.variable values from array values returned.